endpoints/permission/permission_endpoint.ts

```diff
@@ -74,35 +72,25 @@ export default class PermissionEndpoint implements IBaseEndpoint {
   // >>> Rescind >>>
 
   async callEndpoint(
+    this: EndpointThisType<PermissionEndpoint, type.PermissionRequestArgs, Promise<CallEndpointReturnType>>,
     name: string, args: type.PermissionRequestArgs, user: TUser | undefined
   ): Promise<CallEndpointReturnType> {
     const permissionIncludes = this.allowNames.includes(name);
     if (!permissionIncludes) {
       throw new RequestError("EndpointNotFound", `Endpoint permission/${name} does not exist`, 404);
     }
 
-    // Element implicitly has an 'any' type
-    // because expression of type 'string' can't be used to index type 'PermissionEndpoint'.
-    // No index signature with a parameter of type 'string' was found on type 'PermissionEndpoint'.
-    // But it actually can be used
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
     const result: CallEndpointReturnType = await this[name](args, user);
 
     return result;
   }
 
-  async validate(schema: Joi.ObjectSchema, args: type.PermissionRequestArgs): Promise<void> {
-    const validationResult = schema.validate(args);
-    if (validationResult.error) {
-      throw new RequestError("ValidationError", validationResult.error.message, 400);
+  async validate<EType>(schema: Joi.ObjectSchema, args: EType): Promise<EType> {
+    const { error, value } = schema.validate(args);
```

Comment:
```suggestion
    const validData = await schema.validate(args);
```
Вродь так вот можно
И он вродь сам выбрасывать должен будет при ошибке

Reply:
Попробуй и отпиши
Это 50/50 уверенность тут у меня

Reply:
Тут больше как совет был
Иб по моей праките чаще использую с await (попросту меньше кода писать ЪыЪ)
Работает так работает